### PR TITLE
Ember Cli 2.12 upgrade without bower dependency

### PR DESCRIPTION
**Pull request by PoslinskiNet**
# Issues
This PR will close:
- #8
- #11 
- #13

# Tasks:
- [x] ESlint
- [x] ES6
- [x] brocfile removal
- [x] FastBoot support
- [x] Update helpers readme if needed
- [x] Move from IntroJS via bower to introJS via npm
- [x] Readme transition guide
- [x] IntroJS license check & info
- [x] Release the current version (1.x) 
- [x] Release 2.x
- [ ] Publish 2.x via NPM

# Transition guide (release change log)

The biggest change introduced in the version **2.0.0** is exchanging IntroJS bower dependency to NPM equivalent. The new version has support for **FastBoot**, and doesn't cause any deprecation errors when is used with the current version of Ember CLI.

To upgrade from version **1.x** to **2.0.0** you should:

1. If you are only using IntroJS via ember-introjs addon then run: 
   ```bash
   bower uninstall intro.js --save
   ```
   to remove bower dependency, and keep build directory clean, you can reinstall all bower packages with: 
   ```bash
   bower prune & bower install
   ```
2. Update the current version of **ember-introjs**:
   ```bash
   npm install ember-introjs --save-dev
   ```
3. If for some reason, you would like to use IntroJS directly, keep in mind that it is not globally available anymore. You can add it with:
   ```javascript
   import introJs from 'ember-introjs/intro-js';
   ```

### CODE REVIEW

File: addon/components/intro-js.js [hunk 1]
```diff
@@ -25,9 +24,13 @@ var INTRO_JS_OPTIONS = [
 
 var IntroJSComponent = Ember.Component.extend({
 
-  setupIntroJS: Ember.observer('start-if', function(){
+  // setupIntroJS: Ember.observer('start-if', function(){
```
Comment (Exelord) [inline addon/components/intro-js.js]: remove comments

File: package.json [hunk 2]
```diff
@@ -1,40 +1,57 @@
 {
   "name": "ember-introjs",
-  "version": "1.0.0",
-  "description": "The default blueprint for ember-cli addons.",
+  "version": "2.0.0",
+  "description": "Intro.js wrapper component",
```
Comment (Exelord) [inline package.json]: In GitHub description, we have: `An Ember Component for intro.js` Shouldn't we keep that unified?

File: package.json [hunk 3]
```diff
@@ -1,40 +1,57 @@
...
+  "keywords": [
+    "ember-addon"
+  ],
+  "license": "MIT",
```
Comment (Exelord) [inline package.json]: OK, with `MIT` but we should also warn somewhere users that Intro.js is on `AGPL-3.0`

File: package.json [hunk 4]
```diff
@@ -1,40 +1,57 @@
 {
   "name": "ember-introjs",
-  "version": "1.0.0",
-  "description": "The default blueprint for ember-cli addons.",
+  "version": "2.0.0",
+  "description": "Intro.js wrapper component",
+  "keywords": [
+    "ember-addon"
```
Comment (Exelord) [inline package.json]: More meaningful  keywords? :)

File: tests/dummy/app/tests/integration/helpers-test.js [hunk 5]
```diff
@@ -1,42 +0,0 @@
-import startApp from 'dummy/tests/helpers/start-app';
```
Comment (Exelord) [inline tests/dummy/app/tests/integration/helpers-test.js]: You should revert it

File: addon/helpers.js [hunk 6]
```diff
@@ -1,75 +0,0 @@
-import IntroJSComponent from 'ember-introjs/components/intro-js';
```
Comment (Exelord) [inline addon/helpers.js]: You should revert it

File: tests/test-helper.js [hunk 7]
```diff
@@ -1,43 +1,4 @@
-/* globals require, mocha */
```
Comment (Exelord) [inline tests/test-helper.js]: Are you sure we should remove this test-helper file?

File: addon/components/intro-js.js (hunk 1 above)
Comment (PoslinskiNet) [inline addon/components/intro-js.js]: Done

File: addon/helpers.js (hunk 6 above)
Comment (PoslinskiNet) [inline addon/helpers.js]: Done

File: package.json (hunk 2 above)
Comment (PoslinskiNet) [inline package.json]: Done

File: package.json (hunk 4 above)
Comment (PoslinskiNet) [inline package.json]: Done

File: tests/dummy/app/tests/integration/helpers-test.js (hunk 5 above)
Comment (PoslinskiNet) [inline tests/dummy/app/tests/integration/helpers-test.js]: Done

File: tests/test-helper.js (hunk 7 above)
Comment (PoslinskiNet) [inline tests/test-helper.js]: Reverted and moved

**Comment by PoslinskiNet** [comment]
@Exelord I've applied your suggestions and finished tasks. Do you have any further feedback?

**Comment by Exelord** [comment]
@PoslinskiNet  I have also some remarks to the `Transition guide`:
1. You don't have to remove `introjs` from bower deps. You can. It's optional. It won't break the addon.
2. This will not work as expected. It won't remove `introjs` from bower.json 
> Alternatively you can reinstall all bower packages with bower prune & bower install

3. You can just change the version in package.json to 2.0.0 and then type `npm i`... there is no need to `update` the package. What it does is updating npm cache. So, if you are upgrading the package there is no cache :)
4. You should warn! users that we removed `introJs` from the globals. Now you have to import it `import introJs from 'intro-js`. This is the very important change that makes this package incompatible with previous versions.

**Comment by PoslinskiNet** [comment]
@Exelord thanks for the catch. I've updated a guide a little bit.

**Comment by Exelord** [comment]
Nice :) So, now the upgrade path is:
1. `npm install ember-introjs --save-dev` (it will install the latest version and modify package.json)
2. `bower uninstall intro.js --save` (it will remove old intro.js dependency)
3. `import introJs from 'ember-introjs/intro-js'` (import intro in places where it's needed)

**Pull request merged**